Dario api doc

dariodinizg opened the pull request:
Basic Rest API Documentation 
RAML version 1.0

Title: Loan Api

#10

dariodinizg committed:
wip of api documentation

dariodinizg committed:
Included http methods and responses

dariodinizg committed:
refactored code

dariodinizg committed:
included headers on post methods

dariodinizg commented:
I struggled a little with the template and, in the end, I left it aside. Hope its ok.

dariodinizg committed:
Changed 404 description in line 75.

from: Cannot find informations for the given id
to:  Couldn't find informations for the given id

AlexandreProenca commented on dario_loan_api.raml:
```diff
@@ -0,0 +1,78 @@
+#%RAML 1.0
+mediaType: application/json
```
We need specify which protocols this API accept
```suggestion
protocols: [ HTTP, HTTPS ]
```

AlexandreProenca commented on dario_loan_api.raml:
```diff
@@ -0,0 +1,78 @@
+title: Loan API
+version: 1.0
+description: API to manage the loan payments control system from a fintech
+baseUri: http://localhost:3000/
```
In the root description we need define security schema, if we will use JWT
```suggestion
securitySchemes:
  JWT:
    description: Authenticv this to any method that needs a valid JWT to be set.
    type: x-{other}
    describedBy:
      headers:
        Authorization:
          Authorization:
            description: X-AuthToken
            type: string
            required: true
      responses:
        401:
          description: Authentication problem (JWT not set or expired)
    settings:
      roles: []
```

AlexandreProenca commented on dario_loan_api.raml:
```diff
@@ -0,0 +1,78 @@
+description: API to manage the loan payments control system from a fintech
+baseUri: http://localhost:3000/
+types:
+  loan_id:
```
I don't think that we need specify the loan_id, because is a database responsability,  we don't send object id in our request.

AlexandreProenca commented on dario_loan_api.raml:
```diff
@@ -0,0 +1,78 @@
+  loan_id:
+    pattern: ^([0-9]){3}-([0-9]){4}-([0-9]){4}-([0-9]){4}$   
+    example: 000-0000-0000-0000
+  payment:
```
I think the payment not is an object, it is parameters of the payment endpoint.

AlexandreProenca commented on dario_loan_api.raml:
```diff
@@ -0,0 +1,78 @@
+    description: Register a new loan
+    body:
+      type: [loan]
+    headers:     
```
The parameters need passed by body, not in header

AlexandreProenca commented on dario_loan_api.raml:
```diff
@@ -0,0 +1,78 @@
+        headers:
+          Location:         
+            description: '/loans/{loan_id}/'
+      500:
```
We don't need specify 500 error family, it is server responsability.

AlexandreProenca commented on dario_loan_api.raml:
```diff
@@ -0,0 +1,78 @@
+            description: '/loans/{loan_id}/'
+      500:
+        description: Fail to register new loan. Please, try again later    
+  /{loan_id}/payments:
```
we can use just ID
```suggestion
  /{id}/payments:
```

dariodinizg replied:
I think loan id is more accurate.. its not the client id, for example, which could refer to the client registry in their CRM that our api might be integrated. Since this documentation is something to present to the stakeholder, could be interesting give this id the context of our API.
What you think?

AlexandreProenca reviewed (changes requested):
You miss some thinks

Commit:
Update dario_loan_api.raml

Ok! This choice about the JWT wasnt clear to me. I was even reading about Oath...
I will read about JWT then. Thanks!

Co-Authored-By: dariodinizg <43078060+dariodinizg@users.noreply.github.com>

dariodinizg committed:
Suggested changes in PR incorporated
